### PR TITLE
chore: private endpoint juju access

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: N/A

### Overview

- Changes tests to use private endpoints

### Rationale

- Due to constraints of ARM64 nested virtualization, if we deploy ARM64 image builder charm on top of LXD VM (OpenStack runner VM -> LXD VM = 2 nested virt layers), the KVM is not available and the image bulid will fail.

### Juju Events Changes

None.

### Module Changes

None.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: .github/workflows/integration_test.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
       builder-runner-label: X64
 
   integration-tests-arm:
-    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
+    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@feat/private_endpoint
```
Comment (yanksyoon) [inline .github/workflows/integration_test.yaml:24]: Recent merges in craft-cli makes the testing for operator-workflow fail. Use this until that gets resolved.

Comment (yanksyoon) [reply .github/workflows/integration_test.yaml:24]: See https://github.com/canonical/rockcraft/issues/621

File: tests/integration/setup_private_endpoint.sh [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+function load_juju_controller_config(){
+    vault_auth
+    vault read -field=controller_config "${VAULT_SECRET_PATH_COMMON}"/controllers/"${JUJU_CONTROLLER}" | base64 -d - > "${HOME}/.local/share/juju/controllers.yaml"
+}
+
+function load_juju_account_config(){
+    vault_auth
+    # temporarily disable trace
```
Comment (yanksyoon) [inline tests/integration/setup_private_endpoint.sh:75]: Tokens should not show up in console, even if this lives for a short duration.

File: tests/integration/setup_private_endpoint.sh [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	        password: ${PASSWORD}
+	EOF
+}
+
+function switch_juju_model(){
+    # cannot switch if these environment variables are set
+    # by default, the model is not selected and `juju status` will fail
+    unset JUJU_CONTROLLER JUJU_MODEL
+    juju switch "$(juju models --format json | jq -r '.models[0].name')"
+}
```
Comment (yanksyoon) [inline tests/integration/setup_private_endpoint.sh:97]: Changed from POC scriptlet.

File: .github/workflows/integration_test.yaml (hunk 1 above)
Comment (jdkandersson) [reply .github/workflows/integration_test.yaml:24]: Should be fixed now

File: tests/integration/conftest.py [hunk 4]
```diff
@@ -95,31 +149,50 @@ def openstack_clouds_yaml_fixture(pytestconfig: pytest.Config) -> str:
...
 
 @pytest.fixture(scope="module", name="network_name")
-def network_name_fixture(pytestconfig: pytest.Config) -> str:
+def network_name_fixture(pytestconfig: pytest.Config, arch: Literal["amd64", "arm64"]) -> str:
     """Network to use to spawn test instances under."""
-    network_name = pytestconfig.getoption("--openstack-network-name")
-    assert network_name, "Please specify the --openstack-network-name command line option"
+    network_name: str
+    if arch == "arm64":
+        network_name = pytestconfig.getoption("--openstack-network-name")
```
Comment (jdkandersson) [inline tests/integration/conftest.py:156]: these should probably all have the architecture in them

**Review by yhaliaw — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]

**Pull request merged**